### PR TITLE
Add support for create transaction with origin

### PR DESCRIPTION
**Pull request by d-moreira**
This PR adds support for creating a transaction with origin

### CODE REVIEW

**Comment by SandroMachado** [comment]
- [ ] Add a test for the `createTransaction` method at the `UserCardService` with a `TransactionOriginRequest`.

**Comment by d-moreira** [comment]
Updated @SandroMachado.

**Comment by SandroMachado** [comment]
- [ ] Update the documentation.

**Comment by d-moreira** [comment]
Updated @SandroMachado.

**Comment by d-moreira** [comment]
Updated @SandroMachado.

File: README.md [hunk 1]
```diff
@@ -190,9 +190,22 @@ upholdClient.getTickersByCurrency("BTC").then(new PromiseAction<List<Rate>>() {
...
+card.createTransaction(transactionDestinationRequest).then(new PromiseAction<Transaction>() {
+    @Override
+    public void call(Transaction transaction) {
+        // Commit the transaction.
+        transaction.commit();
+    }
+});
+
+// Transaction Origin Request
+TransactionOriginRequest transactionOriginRequest = new TransactionOriginRequest(transactionDenominationRequest, "foo@bar.com");
```
Comment (SandroMachado) [inline README.md]: Replace `foo@bar.com` by `foobar` since usually the origin is an account id and not an email.

File: README.md [hunk 2]
```diff
@@ -190,9 +190,22 @@ upholdClient.getTickersByCurrency("BTC").then(new PromiseAction<List<Rate>>() {
...
+
+card.createTransaction(transactionDestinationRequest).then(new PromiseAction<Transaction>() {
+    @Override
+    public void call(Transaction transaction) {
+        // Commit the transaction.
+        transaction.commit();
+    }
+});
+
+// Transaction Origin Request
```
Comment (SandroMachado) [inline README.md]: `A transaction from an origin.`?

File: README.md [hunk 3]
```diff
@@ -190,9 +190,22 @@ upholdClient.getTickersByCurrency("BTC").then(new PromiseAction<List<Rate>>() {
 
 ```java
 TransactionDenominationRequest transactionDenominationRequest = new TransactionDenominationRequest("1.0", "BTC");
-TransactionRequest transactionRequest = new TransactionRequest(transactionDenominationRequest, "foo@bar.com");
 
-card.createTransaction(transactionRequest).then(new PromiseAction<Transaction>() {
+// Transaction Destination Request
```
Comment (SandroMachado) [inline README.md]: `A transaction to a destination.`?

**Comment by d-moreira** [comment]
Updated @SandroMachado.

File: src/app/src/androidTest/java/com/uphold/uphold_android_sdk/test/integration/model/CardTest.java [hunk 4]
```diff
@@ -185,7 +185,7 @@ public void createTransactionShouldReturnTheTransactionCommitted() throws Except
 
                 card.setUpholdRestAdapter(adapter);
 
-                return card.createTransaction(new TransactionRequest(null, "foobar"), true);
+                return card.createTransaction(new TransactionDestinationRequest(null, "foobar"), true);
```
Comment (SandroMachado) [inline src/app/src/androidTest/java/com/uphold/uphold_android_sdk/test/integration/model/CardTest.java]: Please test all the transactions scenarios.

File: src/app/src/main/java/com/uphold/uphold_android_sdk/model/transaction/TransactionCardOriginRequest.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
+package com.uphold.uphold_android_sdk.model.transaction;
+
+/**
+ * Transaction card origin request model.
+ */
+
+public class TransactionCardOriginRequest extends TransactionOriginRequest {
+
+    private String securityCode;
```
Comment (SandroMachado) [inline src/app/src/main/java/com/uphold/uphold_android_sdk/model/transaction/TransactionCardOriginRequest.java]: Add the `final`.

**Comment by d-moreira** [comment]
Updated @SandroMachado.

File: src/app/src/androidTest/java/com/uphold/uphold_android_sdk/test/integration/service/UserCardServiceTest.java [hunk 6]
```diff
@@ -92,7 +94,7 @@ public void confirmTransactionShouldReturnTheRequest() throws Exception {
     }
 
     @Test
-    public void createTransactionShouldReturnTheRequest() throws Exception {
+    public void createTransactionWithDestinationShouldReturnTheRequest() throws Exception {
```
Comment (SandroMachado) [inline src/app/src/androidTest/java/com/uphold/uphold_android_sdk/test/integration/service/UserCardServiceTest.java]: WDYT of renaming to `createTransactionTransferShouldReturnTheRequest`?

File: src/app/src/androidTest/java/com/uphold/uphold_android_sdk/test/integration/service/UserCardServiceTest.java [hunk 7]
```diff
@@ -101,7 +103,51 @@ public void createTransactionShouldReturnTheRequest() throws Exception {
...
+        });
+
+        Request request = adapter.getRequest();
+
+        Assert.assertEquals(request.getMethod(), "POST");
+        Assert.assertEquals(request.getUrl(), String.format("%s/v0/me/cards/foobar/transactions?commit=false", BuildConfig.API_SERVER_URL));
+    }
+
+    @Test
+    public void createTransactionWithOriginShouldReturnTheRequest() throws Exception {
```
Comment (SandroMachado) [inline src/app/src/androidTest/java/com/uphold/uphold_android_sdk/test/integration/service/UserCardServiceTest.java]: WDYT of renaming to `createTransactionDepositShouldReturnTheRequest`.

File: src/app/src/androidTest/java/com/uphold/uphold_android_sdk/test/integration/service/UserCardServiceTest.java [hunk 8]
```diff
@@ -101,7 +103,51 @@ public void createTransactionShouldReturnTheRequest() throws Exception {
...
+        });
+
+        Request request = adapter.getRequest();
+
+        Assert.assertEquals(request.getMethod(), "POST");
+        Assert.assertEquals(request.getUrl(), String.format("%s/v0/me/cards/foobar/transactions?commit=false", BuildConfig.API_SERVER_URL));
+    }
+
+    @Test
+    public void createTransactionWithSecurityCodeShouldReturnTheRequest() throws Exception {
```
Comment (SandroMachado) [inline src/app/src/androidTest/java/com/uphold/uphold_android_sdk/test/integration/service/UserCardServiceTest.java]: WDYT of renaming to `createTransactionCardDepositShouldReturnTheRequest`?

File: src/app/src/main/java/com/uphold/uphold_android_sdk/model/transaction/TransactionTransferRequest.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
+package com.uphold.uphold_android_sdk.model.transaction;
+
+/**
+ * Transaction destination request model.
+ */
+
+public class TransactionTransferRequest extends TransactionRequest {
```
Comment (SandroMachado) [inline src/app/src/main/java/com/uphold/uphold_android_sdk/model/transaction/TransactionTransferRequest.java]: `implements Serializable` is missing.

File: src/app/src/main/java/com/uphold/uphold_android_sdk/model/transaction/TransactionDepositRequest.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
+package com.uphold.uphold_android_sdk.model.transaction;
+
+/**
+ * Transaction deposit request model.
+ */
+
+public class TransactionDepositRequest extends TransactionRequest {
```
Comment (SandroMachado) [inline src/app/src/main/java/com/uphold/uphold_android_sdk/model/transaction/TransactionDepositRequest.java]: `implements Serializable` is missing.

File: src/app/src/main/java/com/uphold/uphold_android_sdk/model/transaction/TransactionCardDepositRequest.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
+package com.uphold.uphold_android_sdk.model.transaction;
+
+/**
+ * Transaction card deposit request model.
+ */
+
+public class TransactionCardDepositRequest extends TransactionDepositRequest {
```
Comment (SandroMachado) [inline src/app/src/main/java/com/uphold/uphold_android_sdk/model/transaction/TransactionCardDepositRequest.java]: `implements Serializable` is missing.

**Pull request merged**